tests/check_check_tags.c

```diff
@@ -0,0 +1,621 @@
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+#include "../lib/libcompat.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <check.h>
+#include <assert.h>
+#include "check_check.h"
+
+enum {
+    RED,
+    BLUE,
+    PURPLE,
+    YELLOW,
+    BLACK,
+    MAX_TESTS
+} test_ids;
+
+unsigned int test_executed[MAX_TESTS];
+
+static void reset_executed(void)
+{
+    unsigned int i;
+
+    for (i = 0; i < MAX_TESTS; i++)
+    {
+	test_executed[i] = 0;
+    }
+}
+
+START_TEST(red_test1)
+{
+    test_executed[RED] = 1;
+}
+END_TEST
+
+START_TEST(blue_test1)
+{
+    test_executed[BLUE] = 1;
+}
+END_TEST
+
+START_TEST(purple_test1)
+{
+    test_executed[PURPLE] = 1;
+}
+END_TEST
+
+START_TEST(yellow_test1)
+{
+    test_executed[YELLOW] = 1;
+}
+END_TEST
+
+START_TEST(black_test1)
+{
+    test_executed[BLACK] = 1;
+}
+END_TEST
+
+
+static Suite *make_tagged_suite(void)
+{
+    Suite *s;
+
+    TCase *red, *blue, *purple, *yellow, *black;
+
+    s = suite_create("Check Tag Filtering");
+
+    red = tcase_create("Red");
+    tcase_set_tags(red, "Red");
+
+    suite_add_tcase (s, red);
+    tcase_add_test(red, red_test1);
+
+    blue = tcase_create("Blue");
+    tcase_set_tags(blue, "Blue");
+    suite_add_tcase (s, blue);
+    tcase_add_test(blue, blue_test1);
+
+    purple = tcase_create("Purple");
+    tcase_set_tags(purple, "Red Blue");
+    suite_add_tcase (s, purple);
+    tcase_add_test(purple, purple_test1);
+
+    yellow = tcase_create("Yellow");
+    tcase_set_tags(yellow, "Yellow");
+    suite_add_tcase (s, yellow);
+    tcase_add_test(yellow, yellow_test1);
+
+    black = tcase_create("Black");
+    suite_add_tcase (s, black);
+    tcase_add_test(black, black_test1);
+
+    return s;
+}
+
+static   SRunner *sr = NULL;
+
+static void tag_test_setup(void)
+{
+    Suite *s;
+
+    s = make_tagged_suite();
+    sr = srunner_create(s);
+    srunner_set_fork_status(sr, CK_NOFORK);
+}
+
+static void tag_test_teardown (void)
+{
+    srunner_free (sr);
+}
+
+START_TEST(set_get_basic)
+{
+    char ret_str[100];
+    const char *expected_str = "Red Blue Green Yellow";
+    int ret_val;
+    TCase *tc;
+
+    tc = tcase_create("TC");
+    tcase_set_tags(tc, expected_str);
+    ret_val = tcase_get_tags(tc, ret_str, sizeof(ret_str));
```

Comment:
Not sure that a tcase_get_tags() call is necessary. Should this be removed from the API? Do you foresee a compelling usecase for it?